### ISSUE TITLE
TypeError: Cannot mix BigInt and other types, use explicit conversions

### ISSUE THREAD

**Issue by zwang** [rogchap/v8go#92]
I got this when I do something like this:

1. Bind the function from native side:

```
        start := time.Now()
	hrTimeFunc, err := v8go.NewFunctionTemplate(iso, func(info *v8go.FunctionCallbackInfo) *v8go.Value {
		elapsed := time.Since(start).Nanoseconds()
		ret, err := v8go.NewValue(iso, elapsed)
		if err != nil {
			log.Errorf("failed to create v8go NewValue with %d: %v", elapsed, err)
			return nil
		}
		return ret
	})

	if err != nil {
		return err
	}

	err = globalObject.Set("hrtime", hrTimeFunc, v8go.ReadOnly)
```

2 Then run script
`_, err := ctx.RunScript("var result = hrtime() + 1e6;")`

3. Then I got error: "TypeError: Cannot mix BigInt and other types, use explicit conversions"

I checked the code, the reason is because the value returned in hrtime() is actually a BigInt wrapping an int64. V8 does not allow direct calculation between BigInt and other number types. The direct cause of this error is at here:  https://github.com/rogchap/v8go/blob/master/value.go#L67

This can be easily fixed, I just wonder how should I fix it. Why it is not using some code like this at first place? Any concern here?
```
ValuePtr NewValueLong(IsolatePtr iso_ptr, int64_t v) {
  ISOLATE_SCOPE_INTERNAL_CONTEXT(iso_ptr);
  m_value* val = new m_value;
  val->iso = iso;
  val->ctx = ctx;
  val->ptr = Persistent<Value, CopyablePersistentTraits<Value>>(
      iso, Integer::New(iso, v));
  return tracked_value(ctx, val);
}
```

**Comment by rogchap** [comment]
A Javascript Integer Max is 2^53 so for a Go int64 we need to use a BigInt otherwise we may overflow.

`v8go.NewValue()` is a convenience function, maybe we should create explicit functions too ie. `v8go.NewInteger()` 🤔

**Comment by zwang** [comment]
yeah, I figured my solution is only up to 32 bit. it is not a real solution although it will work most of time.

I guess there is no real solution to achieve the two out comes at the same time. 
1. No overflow
2. Easy and intuitive calculation between BigInt and regular numbers. (no forced type conversion in javascript code)

Probably for me, the solution will be to return Int32 in the API above whenever possible.  Hopefully never need to return an Int64.
